libpromises/evalfunction.c

```diff
@@ -509,67 +509,160 @@ static JsonElement* VarNameOrInlineToJson(EvalContext *ctx, const FnCall *fp, co
     return vardata;
 }
 
-static Rlist *GetHostsFromLastseenDB(Item *addresses, time_t horizon, bool return_address, bool return_recent)
+/*********************************************************************/
+
+static FnCallResult FnCallAnd(EvalContext *ctx,
+                              ARG_UNUSED const Policy *policy,
+                              ARG_UNUSED const FnCall *fp,
+                              const Rlist *finalargs)
+{
+
+    for (const Rlist *arg = finalargs; arg; arg = arg->next)
+    {
+        SyntaxTypeMatch err = CheckConstraintTypeMatch(fp->name, arg->val, CF_DATA_TYPE_STRING, "", 1);
+        if (err != SYNTAX_TYPE_MATCH_OK && err != SYNTAX_TYPE_MATCH_ERROR_UNEXPANDED)
+        {
+            FatalError(ctx, "Function '%s', %s", fp->name, SyntaxTypeMatchToString(err));
+        }
+    }
+
+    for (const Rlist *arg = finalargs; arg; arg = arg->next)
+    {
+        if (!IsDefinedClass(ctx, RlistScalarValue(arg)))
+        {
+            return FnReturnContext(false);
+        }
+    }
+
+    return FnReturnContext(true);
+}
```

Comment:
Why did you move FnCallAnd? If there is a good reason, do it in a separate commit and explain why.

Reply:
The reason is mostly cohesion. The GetHostsFromLastseenDB was above FnCallAnd and yet it wasnt used there. So i figured i would consolidate all the FnCallHostsSeen functions by moving that bellow And. I will revert it for now.